snooty/types.py

```diff
@@ -427,6 +440,29 @@ def merge_facets(
 
         return merged_facets
 
+    @staticmethod
+    def get_facet_display_name(
+        parent_facets: List[Facet], unparsed_facet: Dict[str, Any]
+    ) -> str:
+        try:
+            if unparsed_facet["category"] == "version":
+                return str(unparsed_facet["value"])
+            taxonomy_ref = asdict(taxonomy.TaxonomySpec.get_taxonomy())
+            for parent in parent_facets:
+                options_list = taxonomy_ref[parent.category] or []
+                taxonomy_ref = [x for x in options_list if x["name"] == parent.value][0]
+            options_list = taxonomy_ref[unparsed_facet["category"]]
+            taxonomy_ref = [
+                x for x in options_list if x["name"] == str(unparsed_facet["value"])
+            ][0]
+            return str(
+                taxonomy_ref["display_name"]
+                or str(taxonomy_ref["name"]).replace(r"\_", " ").title()
+            )
+        except Exception:
+            logger.warning(f"Display name not found for invalid facet {unparsed_facet}")
```

Comment:
(Non-blocking question) Is there a reason why we use `logger` here instead of adding a `diagnostic` to help find the invalid facet? Is it just more convenient to use the logger in this case?

Reply:
thought this might be noisy as we already have a check for the [project's facets.toml](https://github.com/mongodb/snooty-parser/blob/main/snooty/types.py#L391) file as well as the [pages facets directives](https://github.com/mongodb/snooty-parser/blob/main/snooty/parser.py#L601)